subsys/usb/class/cdc_acm.c

```diff
@@ -84,9 +84,7 @@ LOG_MODULE_REGISTER(usb_cdc_acm);
 #define ACM_IF0_STRING			"ACM-CDC"
 
 struct usb_cdc_acm_config {
-#ifdef CONFIG_USB_COMPOSITE_DEVICE
 	struct usb_association_descriptor iad_cdc;
-#endif
```

Comment:
Why Interface Association Descriptor has to be used with Windows if there is only one instance of the function (e.g. CDC ACM)? Where does it come from?

Reply:
One instance but CDC ACM uses two interfaces. Without it Windows detects two separate devices.

Reply:
Windows OS version? Is it `samples/subsys/usb/cdc_acm` ?

Reply:
Windows 7 SP1 x64. Yes.

Reply:
Confirm that it does not work without IAD on Windows 7. But I still do not see a reason to include IAD in single function configuration, especially since Union Interface Functional Descriptor is there to group the interfaces.

Does Windows 10 OS works without any modifications?
Are these modification only necessary for earlier Windows OS versions?
What about macOS, *BSD ? :-)